File: java/lab5/src/commands/HelpCommand.java
package src.commands;

import src.baseabstractions.Command;
import src.basecollection.CollectionManager;
import src.baseobjects.Product;

/**
 * Command to display help information for all available commands.
 *
 * @param <K> the type of keys used in the collection, must be Comparable
 * @param <V> the type of values stored in the collection, must extend Product
 */
public class HelpCommand<K extends Comparable<K>, V extends Product> extends Command<K, V> {

    /**
     * Constructs a HelpCommand with the specified collection manager.
     *
     * @param collectionManager the collection manager to operate on
     */
    public HelpCommand(CollectionManager<K, V> collectionManager) {
        super(collectionManager);
    }

    /**
     * Executes the help command.
     * Displays descriptions of all available commands.
     *
     * @param args command arguments (expects exactly one argument - the command
     *             name)
     * @throws IllegalArgumentException if the number of arguments is incorrect
     */
    public void execute(String[] args) {
        if (args.length == 1) {
            getCollectionManager().help();
        } else
            throw new IllegalArgumentException("Number of arguments is wrong.");
    }

    /**
     * Returns a description of the command.
     *
     * @return "help : вывести справку по доступным командам" - command description
     *         in Russian
     */
    public String getDescription() {
        return "help : вывести справку по доступным командам";
    }
}

File: java/lab6/client/src/main/kotlin/commands/HelpCommand.kt
package commands

import abstractions.AbstractCommandManager
import abstractions.Command
import objects.Product


class HelpCommand<K : Comparable<K>, V : Product>(
    private val commandManager: AbstractCommandManager<K, V>
) : Command<K, V>() {

    override val description = "help : вывести справку по доступным командам"

    override val isShouldBeSent = false

    override fun start(args: Array<String>) {
        if (args.size != 1) {
            throw IllegalArgumentException("Number of arguments is wrong.")
        }
        commandManager.commands.forEach { (_, command) -> println(command.description) }
    }

    override fun finish() {}
}